### PR TITLE
Map MODS to cocina before updating.

### LINKED ISSUE #891: Map MODS to cocina before updating
Instead of using the `update_mods` API call, use the functions in cocina-models to convert to cocina first.

```
Cocina::Models::Description.new(
  Cocina::Models::Mapping::FromMods::Description.props(mods: mods_ng, druid: cocina.externalIdentifier, label: cocina.label)
)
```

Do that here:
https://github.com/sul-dlss/common-accessioning/blob/5b75f0ca9a68f84e13c86a62625c67e45f0658fc/lib/robots/dor_repo/accession/descriptive_metadata.rb#L18-L19

This is similar to the work in: https://github.com/sul-dlss/argo/pull/3614

### PR DESCRIPTION
**Pull request by justinlittman**
closes #891

## Why was this change made? 🤔
Allow retiring the mods update endpoint.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit

### CODE REVIEW

**Comment by justinlittman** [comment]
P.S. I made additions to shared configs.

File: config/settings.yml [hunk 1]
```diff
@@ -32,3 +32,5 @@ workflow:
 
 redis:
   url: ~
+
+purl_url: 'https://purl-example.stanford.edu'
```
Comment (jcoyne) [inline config/settings.yml:36]: Has this change been made in shared configs for the 3 environments?

File: lib/robots/dor_repo/base.rb [hunk 2]
```diff
@@ -5,6 +5,11 @@ module DorRepo
     class Base
       include LyberCore::Robot
 
+      def initialize(...)
+        super
+        Cocina::Models::Mapping::Purl.base_url = Settings.purl_url
```
Comment (jcoyne) [inline lib/robots/dor_repo/base.rb:10]: Can this be in the DescriptiveMetadata robot or in config/boot.rb ?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: config/settings.yml (hunk 1 above)
Comment (justinlittman) [reply config/settings.yml:36]: Yes.

File: lib/robots/dor_repo/base.rb (hunk 2 above)
Comment (justinlittman) [reply lib/robots/dor_repo/base.rb]: Ahh, I did not know about `boot.rb`. Fixed.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**